### PR TITLE
Fix 13823: Validate and Parse gracefully IngestionPipeline

### PR DESCRIPTION
**Pull request by OnkarVO7**
### Describe your changes:

Fixes https://github.com/open-metadata/OpenMetadata/issues/13823

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/api/parser.py [hunk 1]
```diff
@@ -434,3 +437,38 @@ def parse_test_connection_request_gracefully(
...
+
+    :param config_dict: JSON ingestion pipeline config
+    :return:Ingestion Pipeline config or scoped error
+    """
+
+    try:
+        ingestion_pipeline = IngestionPipeline.parse_obj(config_dict)
+        return ingestion_pipeline
+
+    except ValidationError:
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/api/parser.py:458]: should we take care of the `serviceConnection`? Not only the `sourceConfig`

**Review by pmbrull — CHANGES_REQUESTED** [review]

File: ingestion/src/metadata/ingestion/api/parser.py (hunk 1 above)
Comment (pmbrull) [reply ingestion/src/metadata/ingestion/api/parser.py:458]: I believe that's where we get also a lot of noise, for example when we configure an URI without the scheme or something

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/api/parser.py:458]: as discussed, that was the wrong comment. We'll be adding it to the run_automations endpoint though

File: ingestion/src/metadata/ingestion/api/parser.py [hunk 2]
```diff
@@ -400,37 +406,84 @@ def parse_workflow_config_gracefully(
     raise ParsingConfigurationError("Uncaught error when parsing the workflow!")
 
 
-def parse_test_connection_request_gracefully(
```
Comment (OnkarVO7) [inline ingestion/src/metadata/ingestion/api/parser.py:403]: Removing the method since it wasn't being used anywhere except for tests.

**Review by pmbrull — APPROVED** [review]

**Pull request merged**